Lib/venv/__init__.py

```diff
@@ -116,7 +116,7 @@ def create_if_needed(d):
             elif os.path.islink(d) or os.path.isfile(d):
                 raise ValueError('Unable to create directory %r' % d)
 
-        if os.pathsep in env_dir:
+        if os.pathsep in str(env_dir):
```

Comment:
```suggestion
        if os.pathsep in os.fspath(env_dir):
```
Stringifying isn't necessarily the right solution for all PathLikes.